src/DocumentEditor.ts

```diff
@@ -4,154 +4,79 @@
-        this.recoveredDocsFolder = path.join(context.extensionPath, 'recoveredDocs');
-
-        this.promptForRecoveredDocs();
-    }
-
-    public async showDocument(docNode: IDocumentNode): Promise<void> {
-        // Prompt to update opened doc if it's dirty
-        if (this.lastOpenedDocNode && this.localDoc) {
-            // soft-copy the node and doc to avoid race conditions
-            const doc: vscode.TextDocument = this.localDoc;
-            const node: IDocumentNode = this.lastOpenedDocNode;
-
-            if (doc.isDirty) {
-                const message: string = `Your changes to document '${this.lastOpenedDocNode.label}' will be lost. Update to Azure?`;
-                const result: string | undefined = await vscode.window.showWarningMessage(message, DialogBoxResponses.Yes, DialogBoxResponses.No);
-                if (result === DialogBoxResponses.Yes) {
-                    await this.udpateDocumentToNode(node, doc);
-                } else if (result === undefined) {
-                    throw new UserCancelledError();
-                }
-            }
-        }
+    public async showDocument(docNode: IDocument, fileName: string): Promise<void> {
+        const localDocPath = path.join(os.tmpdir(), randomUtils.getRandomHexString(12), fileName);
+        await fse.ensureFile(localDocPath);
 
-        await fse.ensureFile(this.localDocPath);
-        this.localDoc = await vscode.workspace.openTextDocument(this.localDocPath);
-        await vscode.window.showTextDocument(this.localDoc);
-        this.lastOpenedDocNode = docNode;
-        this.isLocalDocOpen = true;
-        await this.updateEditor(docNode.data);
+        const document = await vscode.workspace.openTextDocument(localDocPath);
+        this.fileMap[localDocPath] = [document, docNode];
+        const textEditor = await vscode.window.showTextDocument(document);
+        await this.updateEditor(docNode.data, textEditor);
     }
 
-    public async updateLastDocument(): Promise<void> {
-        if (this.lastOpenedDocNode && this.localDoc) {
-            await this.udpateDocumentToNode(this.lastOpenedDocNode, this.localDoc);
-        } else {
-            throw new Error('You must select a Document in the CosmosDB explorer before updating to Azure.');
-        }
+    public async updateLastDocument(doc): Promise<void> {
+        const filePath = Object.keys(this.fileMap).find((filePath) => path.relative(doc.fsPath, filePath) === '');
+        await this.udpateDocumentToNode(this.fileMap[filePath][1], this.fileMap[filePath][0]);
     }
 
     public async dispose(): Promise<void> {
-        if (this.recoveredFileName) {
-            await fse.ensureDir(this.recoveredDocsFolder);
-            await fse.move(this.localDocPath, path.join(this.recoveredDocsFolder, this.recoveredFileName));
-        } else {
-            await fse.unlink(this.localDocPath);
-        }
+        Object.keys(this.fileMap).forEach(async (key) => await fse.unlink(key));
```

Comment:
Did you figure out how to fix the error message when you close and re-open VS Code? If not, file an issue so we don't forget

Reply:
Filed one.